src/components/responsive-element.tsx

```diff
@@ -4,28 +4,49 @@ import classnames from "classnames";
 type ResponsiveElementTagName = "h1" | "h2" | "h3" | "h4" | "p";
 
 type ResponsiveElementInfo = {
-  desktop: ResponsiveElementTagName;
-  touch: ResponsiveElementTagName;
   children: React.ReactNode;
+  mobile?: ResponsiveElementTagName;
+  tabletOnly?: ResponsiveElementTagName;
+  desktopOnly?: ResponsiveElementTagName;
+  widescreenOnly?: ResponsiveElementTagName;
+  touch?: ResponsiveElementTagName;
+  tablet?: ResponsiveElementTagName;
+  desktop?: ResponsiveElementTagName;
+  widescreen?: ResponsiveElementTagName;
+  fullhd?: ResponsiveElementTagName;
   className?: string;
 };
 
-const ResponsiveElement = ({
-  desktop,
-  touch,
-  children,
-  className,
-}: ResponsiveElementInfo) => {
-  const Desktop = desktop as keyof JSX.IntrinsicElements;
-  const Touch = touch as keyof JSX.IntrinsicElements;
+const ResponsiveElement = (props: ResponsiveElementInfo): JSX.Element => {
+  const { children, className, ...sizeTags } = props;
+
+  if (Object.keys(sizeTags).length < 2)
+    throw new Error("At least two sizes must be given.");
+
+  // https://bulma.io/documentation/helpers/visibility-helpers/#hide
+  const hiddenClasses = {
```

Comment:
small: could move this outside of the component so that it's a shared single variable instead of a new one for each instance

Reply:
good idea, thanks!